packages/babel-parser/src/parser/expression.js

```diff
@@ -2234,9 +2235,34 @@ export default class ExpressionParser extends LValParser {
       );
     }
 
+    if (!this.scope.inFunction && !this.options.allowAwaitOutsideFunction) {
+      if (
+        this.hasPrecedingLineBreak() ||
+        // All the following expressions are ambiguous:
+        //   await + 0
+        //   await - 0
+        //   await ( 0 )
+        //   await [ 0 ]
+        //   await / 0 /u
+        this.match(tt.plusMin) ||
+        this.match(tt.parenL) ||
+        this.match(tt.bracketL) ||
+        this.match(tt.backQuote) ||
```

Comment:
Awesome! I missed ``await ` 0 ` ``.

I come up with a new case: If one turns on `v8intrinsic`, the following could be ambiguous, too.

```js
await %a()
```